drivers/hmis/spec/system/hmis/assessment_definitions_spec.rb

```diff
@@ -0,0 +1,171 @@
+    let!(:old_definition) do
+      fd = create(:custom_assessment_with_custom_fields_and_rules, identifier: definition.identifier, title: 'Previous Very Custom Assessment', data_source: ds1, version: 0, status: :retired)
+      fd.definition['item'][0]['item'] << { 'type': 'DISPLAY', 'link_id': 'old_message', 'text': 'Text on old form' }
+      fd.save!
+      fd
+    end
+
+    before(:each) do
+      create(:hmis_form_instance, definition: definition, entity: p1) # enable the form in p1
+      sign_in(hmis_user)
+      disable_transitions
+
+      visit "/client/#{c1.id}/enrollments/#{e1.id}/assessments"
+    end
+
+    context 'with no existing assessment' do
+      it 'creates a new assessment with the most recent form version' do
+        click_button 'New Assessment'
+        click_link definition.title
+        mui_date_select 'Assessment Date', date: today
+
+        expect do
+          click_button 'Submit'
+          assert_text "#{c1.brief_name} Assessments" # wait until we're back on the assessment table
+        end.to change(e1.custom_assessments, :count).by(1).
+          and change(definition.form_processors, :count).by(1)
+      end
+    end
+
+    context 'with an existing assessment that was created with an old definition' do
+      let!(:assessment) { create(:hmis_custom_assessment, definition: old_definition, enrollment: e1, assessment_date: today) }
+
+      it 'opens the assessment with the original form version' do
+        click_link old_definition.title
+
+        assert_text old_definition.title
+        assert_text 'Assessment Date'
+        assert_text assessment.assessment_date.strftime('%m/%d/%Y')
+        assert_text 'Text on old form'
+      end
+
+      it 'upgrades to the newer version when you unlock for editing' do
+        click_link old_definition.title
+
+        assert_text old_definition.title
+
+        # unlock the assessment should upgrade to the newer form
+        click_button('Unlock Assessment', match: :first)
+        assert_no_text old_definition.title
+        assert_text definition.title
+        expect(page).to have_field('Assessment Date', with: assessment.assessment_date.strftime('%m/%d/%Y'))
+        assert_no_text 'Text on old form'
+      end
+    end
+
+    context 'with an existing WIP assessment that was created with an old definition' do
+      let!(:wip_assessment) { create(:hmis_custom_assessment, definition: old_definition, wip: true, values: { 'assessment_date': today.strftime('%Y-%m-%d') }, enrollment: e1, assessment_date: today) }
+
+      it 'uses old form version for editing' do
+        click_link old_definition.title
+        assert_text old_definition.title
+        expect(page).to have_field('Assessment Date', with: wip_assessment.assessment_date.strftime('%m/%d/%Y'))
+        assert_text 'Text on old form'
+      end
+    end
+  end
+
+  context 'Performing Household Intake assessments with various definition versions' do
```

Comment:
I would appreciate feedback on the structuring of this test. I set up one household, where each member has different states, and tested each member in a different test case.

Reply:
I think this structure is good, especially if it's important that we test the new/old form definition behavior specifically in the context of a multi-member household.

If not, then maybe the test could be simplified and moved to a separate file where the focus is on testing the multi-member household intake process, rather than the old/new form definition behavior.

But if it is important, or if we just want to keep the existing test that you've written (because code wins 🚀 ), then it could be good to add a short comment explaining why the test is testing these 2 sorta-kinda distinct concepts.

Reply:
Okay, thanks. I do think it's important that the household has a variety of assessment setups, so we can be sure it works in that situation. I added a couple comments to that effect!